### PR TITLE
[ENH] `Exponential` distribution

### PR DESCRIPTION
**Pull request by ShreeshaM07**
#### What does this implement/fix? Explain your changes.

This interfaces the `Exponential` distribution using the `_ScipyAdapter`.

#### Does your contribution introduce a new dependency? If yes, which one?
No

#### Did you add any tests for the change?
Yes

##### For all contributions
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [x] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
(https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

File: skpro/distributions/exponential.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    def __init__(self, mu, scale, index=None, columns=None):
+        self.mu = mu
+        self.scale = scale
+
+        super().__init__(index=index, columns=columns)
+
+    def _get_scipy_object(self) -> rv_continuous:
+        return expon
+
+    def _get_scipy_param(self) -> dict:
```
Comment (VascoSch92) [inline skpro/distributions/exponential.py:47]: return type here is a `dict`, but then you are returning a `tuple` composed of a `list` and a `dict`

File: skpro/distributions/exponential.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    scale :math:`b`, such that the pdf is
+
+    .. math:: f(x) = \exp\left(-\frac{x - \mu}{b}\right)
+
+    The mean :math:`\mu` is represented by the parameter ``mu``,
+    and the scale :math:`b` by the parameter ``scale``.
+
+    Parameter
+    ---------
+    mu : float or array of float (1D or 2D)
```
Comment (fkiraly) [inline skpro/distributions/exponential.py:25]: I would give this a single parameter, `rate`. The `scipy` parameter `scale` is then the inverse of `rate`.

File: skpro/distributions/exponential.py (hunk 1 above)
Comment (ShreeshaM07) [reply skpro/distributions/exponential.py:49]: Indeed the return type must be changed.

File: skpro/distributions/exponential.py (hunk 2 above)
Comment (ShreeshaM07) [reply skpro/distributions/exponential.py:25]: I think the exponential distribution takes 2 parameters `loc` and `scale`. If I am not wrong how would having only 1 parameter `rate` work as the distribution is technically `e^-((x-loc)/scale)`.

File: skpro/distributions/__init__.py [hunk 3]
```diff
@@ -21,12 +21,14 @@
     "TDistribution",
     "Uniform",
     "Weibull",
+    "Exponential",
```
Comment (fkiraly) [inline skpro/distributions/__init__.py:24]: can you put this to be alphabetically ordered?

File: docs/source/api_reference/distributions.rst [hunk 4]
```diff
@@ -43,6 +43,7 @@ Continuous support
     Normal
     TDistribution
     Weibull
+    Exponential
```
Comment (fkiraly) [inline docs/source/api_reference/distributions.rst:46]: can you put this to be alphabetically ordered?

File: skpro/distributions/exponential.py (hunk 2 above)
Comment (fkiraly) [reply skpro/distributions/exponential.py:25]: I would suggest, let's use the same parameters as on wikipedia:

https://en.wikipedia.org/wiki/Exponential_distribution

We cannot use `lambda` as it is a reserved python keyword, so `rate` is the next best thing as it is consistent with Poisson process terminology.

**Comment by ShreeshaM07** [comment]
I've made the changes accordingly please let me know if anything else needs to be done. I shall create PR with the updated `ngboost` files as well.

**Review by fkiraly — APPROVED** [review]
Looks good!

**Pull request merged**